server/src/main/java/org/apache/lucene/index/BaseStarTreeBuilder.java

```diff
@@ -268,21 +262,18 @@ long[] getStarTreeDimensionsFromSegment() throws IOException {
     private Object[] getStarTreeMetricsFromSegment() throws IOException {
         Object[] metrics = new Object[numMetrics];
         for (int i = 0; i < numMetrics; i++) {
-            // Ignore the column for COUNT aggregation function
-            DocIdSetIterator metricStatReader = metricAggregationDescriptors.get(i).getMetricStatReader();
+            DocIdSetIterator metricStatReader = metricAggregatorInfos.get(i).getMetricStatReader();
             if (metricStatReader != null) {
                 try {
                     metricStatReader.nextDoc();
                 } catch (IOException e) {
                     logger.error("unable to iterate to next doc", e);
-                } catch (NullPointerException e) {
-                    logger.error("metric does not have an associated reader", e);
-                    throw new IllegalStateException("metric should have a valid associated reader", e);
+                    throw e;
```

Comment:
I see in Lucene that either UnCheckedIOException is thrown or runtimeException is thrown. Maybe lets follow the same unless you have seem some other patterns.